resources/Microsoft.Windows.Developer/Microsoft.Windows.Developer.psm1

```diff
@@ -800,6 +799,73 @@ class WindowsCapability {
         }
     }
 }
+
+[DSCResource()]
+class NetConnectionProfile {
+    # Key required. Do not set.
+    [DscProperty(Key)]
+    [string]$SID
+
+    [DscProperty(Mandatory)]
+    [string]$InterfaceAlias
+
+    [DscProperty(Mandatory)]
+    [string]$NetworkCategory
+
+    [NetConnectionProfile] Get() {
+        $currentState = [NetConnectionProfile]::new()
+
+        $netConnectionProfile = Get-NetConnectionProfile -InterfaceAlias $this.InterfaceAlias -ErrorAction SilentlyContinue
+        if ($null -eq $netConnectionProfile) {
+            throw "No network profile found for interface alias '$($this.InterfaceAlias)'"
+        }
+
+        $currentState.InterfaceAlias = $this.InterfaceAlias
+        $currentState.NetworkCategory = $netConnectionProfile.NetworkCategory
+        return $currentState
+    }
+
+    [bool] Test() {
+        $currentState = $this.Get()
+        return $currentState.NetworkCategory -eq $this.NetworkCategory
+    }
+
+    [void] Set() {
+        if (-not $this.Test()) {
+            Set-NetConnectionProfile -InterfaceAlias $this.InterfaceAlias -NetworkCategory $this.NetworkCategory
+        }
+    }
+}
+
+[DSCResource()]
+class NetConnectionProfileInfo {
```

Comment:
This is less interesting due to a [known DSC bug that blocks asserts from taking parameters](https://github.com/PowerShell/DSC/issues/381). I am planning to break this into its own PR due to this.